File: qb-engineer-server/qb-engineer.core/Models/SalesOrderListItemModel.cs
namespace QBEngineer.Core.Models;

public record SalesOrderListItemModel(
    int Id,
    string OrderNumber,
    int CustomerId,
    string CustomerName,
    string Status,
    string? CustomerPO,
    int LineCount,
    decimal Total,
    DateTimeOffset? RequestedDeliveryDate,
    DateTimeOffset CreatedAt);

File: qb-engineer-ui/src/app/features/sales-orders/models/sales-order-list-item.model.ts
export interface SalesOrderListItem {
  id: number;
  orderNumber: string;
  customerId: number;
  customerName: string;
  status: string;
  customerPO: string | null;
  lineCount: number;
  total: number;
  requestedDeliveryDate: string | null;
  createdAt: string;
}
